src/Orchard.Web/Modules/Orchard.Setup/Services/WebconfigOverrides.cs

```diff
@@ -0,0 +1,39 @@
+﻿using System.Configuration;
+
+namespace Orchard.Setup.Services
+{
+
+    public class WebconfigOverrides
+    {
+        private const string AppSettingDefaultDataProvider = "defaultProvider";
+        private const string AppSettingDefaultRecipe = "defaultRecipe";
+        private const string ConnectionStringDefaultConnectionString = "defaultConnection";
+
+        public string DataProvider { get; set; }
+        public string DataConnectionString { get; set; }
+        public string DefaultRecipe { get; set; }
+
+        public static WebconfigOverrides Load()
+        {
+            var result = new WebconfigOverrides();
+
+            if (ConfigurationManager.ConnectionStrings[ConnectionStringDefaultConnectionString] != null)
+            {
+                result.DataConnectionString = ConfigurationManager.ConnectionStrings[ConnectionStringDefaultConnectionString].ConnectionString;
```

Comment:
@sebastienros how can we get the tenant name? Should we inject any dependency to be able to get this value?

The problem with the tenantName prefix approach is that it will always fallback to the setting without the tenant name, because the current ARM template is only configuring the "defaultConnection" connection string.

Perhaps a better approach would be to skip the default connection string configuration is tenantName != "Default". I think this should be also performed for the Media Services module default configuration as well.

@sebastienros We will wait for your input on this feedback. Thanks!